doc/introduction/circuits.rst

```diff
@@ -58,18 +58,19 @@ Quantum functions are a restricted subset of Python functions, adhering to the f
 constraints:
 
 * The quantum function accepts classical inputs, and consists of
-  :doc:`quantum operators <operations>` or sequences of operators called :doc:`templates`,
-  using one instruction per line.
+  :doc:`quantum operators <operations>` or sequences of operators called :doc:`templates`.
 
 * The function can contain classical flow control structures such as ``for`` loops or ``if`` statements.
 
 * The quantum function must always return either a single or a tuple of
-  *measured observable values*, by applying a :doc:`measurement function <measurements>`
-  to a :ref:`qubit observable <intro_ref_ops_qobs>` or :ref:`continuous-value observable <intro_ref_ops_cvobs>`.
+  *measurement values*, by applying a :doc:`measurement function <measurements>`
+  to the qubit register. The most common example is to measure the expectation value of
+  a :ref:`qubit observable <intro_ref_ops_qobs>` or
+  :ref:`continuous-value observable <intro_ref_ops_cvobs>`.
 
 .. note::
 
-    Quantum functions can only be evaluated on a device from within a QNode.
+    Quantum functions are evaluated on a device from within a QNode.
```

Comment:
They typically are executed via QNodes, and it is the user-facing way to do it. But it is not the only one.